src/clj/runbld/notifications.clj

```diff
@@ -0,0 +1,37 @@
+(ns runbld.notifications
+  (:require [runbld.schema :refer :all]
+            [pallet.thread-expr :refer [when->]]
+            [clojure.string :as str]
+            [runbld.util.date :as date]
+            [schema.core :as s]))
+
+(defn strip-out-runbld [src]
+  (let [runbld-shebang (fn [line]
+                         (not
+                          (re-find #"^#!.*runbld" line)))]
+    (->> src
+         java.io.StringReader.
+         clojure.java.io/reader
+         line-seq
+         (filter runbld-shebang)
+         (interpose "\n")
+         (apply str))))
+
+
+(s/defn make-context :- NotifyCtx
+  [opts build failure]
+  (-> build
+      (update-in [:process :cmd] #(str/join " " %))
+      (update-in [:process :cmd-source] strip-out-runbld)
+      (update-in [:process :args] #(str/join " " %))
+      (assoc-in [:process :took-human]
+                (date/human-duration
+                 (/ (-> build :process :took) 1000)))
+      (update-in [:version :hash]
+                 #(->> % (take 7) (apply str)))
+      (assoc-in [:process :failed]
+                (= "FAILURE" (-> build :process :status)))
```

Comment:
Future todo, maybe make status `:failure` (keyword enum) instead of a string? (not that it's this PR's job to do that)